string name
time current_time